src/main/java/com/spikes2212/command/genericsubsystem/commands/MovePIDSpeedController.java

```diff
@@ -0,0 +1,115 @@
+     * @param maxSpeed        the maximum speed of the subsystem
+     * @param minSpeed        the minimum speed of the subsystem
+     * @param setpoint        the setpoint which the PID loop should reach
+     * @param canMove         a predicate that returns true for speeds at which the subsystem can move
+     */
+    public MovePIDSpeedController(Subsystem subsystem, PIDSpeedController speedController, Supplier<Double> maxSpeed,
+                                  Supplier<Double> minSpeed, Supplier<Double> setpoint, Predicate<Double> canMove) {
+        addRequirements(subsystem);
+        this.speedController = speedController;
+        this.maxSpeed = maxSpeed;
+        this.minSpeed = minSpeed;
+        this.setpoint = setpoint;
+        this.canMove = canMove;
+    }
+
+    /**
+     * A constructor that uses a {@link GenericSubsystem}.
+     *
+     * @param genericSubsystem the subsystem this command requires
+     * @param speedController  the {@link PIDSpeedController} this command runs a PID loop on
+     * @param setpoint         the setpoint which the PID loop should reach
+     */
+    public MovePIDSpeedController(GenericSubsystem genericSubsystem, PIDSpeedController speedController,
+                                  Supplier<Double> setpoint) {
+        this(genericSubsystem, speedController, genericSubsystem.maxSpeed, genericSubsystem.minSpeed, setpoint, genericSubsystem::canMove);
+    }
+
+    /**
+     * A partial constructor, omitting {@code canMove}. Assumes the subsystem can always move at any speed between
+     * {@code maxSpeed} and {@code minSpeed}.
+     *
+     * @param subsystem       the subsystem this command requires
+     * @param speedController the {@link PIDSpeedController} this command runs a PID loop on
+     * @param maxSpeed        the maximum speed of the subsystem
+     * @param minSpeed        the minimum speed of the subsystem
+     * @param setpoint        the setpoint which the PID loop should reach
+     */
+    public MovePIDSpeedController(Subsystem subsystem, PIDSpeedController speedController, Supplier<Double> maxSpeed,
+                                  Supplier<Double> minSpeed, Supplier<Double> setpoint) {
+        this(subsystem, speedController, maxSpeed, minSpeed, setpoint, v -> true);
+    }
+
+    @Override
+    public void initialize() {
+        speedController.configureLoop(maxSpeed, minSpeed);
+    }
+
+    @Override
+    public void execute() {
+        speedController.pidSet(setpoint.get());
+    }
+
+    @Override
+    public void end(boolean interrupted) {
+        speedController.finish();
+    }
+
+    @Override
+    public boolean isFinished() {
+        if (!speedController.onTarget(setpoint.get())) {
+            lastTimeNotOnTarget = Timer.getFPGATimestamp();
+        }
+
+        return !canMove.test(speedController.get()) ||
+                Timer.getFPGATimestamp() - lastTimeNotOnTarget >= speedController.getWaitTime();
```

Comment:
isFinished should return true if   Timer.getFPGATimestamp() - lastTimeNotOnTarget is greater than waitTime because if waitTime equals zero the command will stop immediately